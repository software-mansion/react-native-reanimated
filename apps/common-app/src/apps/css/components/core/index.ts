export { default as Text, type TextProps } from './Text';
