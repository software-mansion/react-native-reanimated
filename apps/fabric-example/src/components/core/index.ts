export { default as Text } from './Text';
