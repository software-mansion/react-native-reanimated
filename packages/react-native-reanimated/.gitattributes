plugin/build/* linguist-generated=true

.yarn/releases/** binary
