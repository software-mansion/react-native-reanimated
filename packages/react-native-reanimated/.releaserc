{
  "extends": [
    "../../.releaserc"
  ],
  "branches": [
    "main",
    "[0-9]+\\.[0-9]+-stable",
    "@matipl01/github-release-action"
  ],
  "tagFormat": "reanimated-${version}",
  "name": "React Native Reanimated"
}
