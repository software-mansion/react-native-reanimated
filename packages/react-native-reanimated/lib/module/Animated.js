'use strict';

export { ReanimatedFlatList as FlatList } from './component/FlatList';
export { AnimatedImage as Image } from './component/Image';
export { AnimatedScrollView as ScrollView } from './component/ScrollView';
export { AnimatedText as Text } from './component/Text';
export { AnimatedView as View } from './component/View';
export { addWhitelistedNativeProps, addWhitelistedUIProps } from './ConfigHelper';
export { createAnimatedComponent } from './createAnimatedComponent';
//# sourceMappingURL=Animated.js.map