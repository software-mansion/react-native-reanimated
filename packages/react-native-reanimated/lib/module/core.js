'use strict';

import { controlEdgeToEdgeValues, isEdgeToEdge } from 'react-native-is-edge-to-edge';
import { createSerializable } from 'react-native-worklets';
import { logger, ReanimatedError } from "./common/index.js";
import { ReanimatedModule } from './ReanimatedModule';
import { SensorContainer } from "./SensorContainer.js";
export { startMapper, stopMapper } from "./mappers.js";
export { makeMutable } from "./mutables.js";
const EDGE_TO_EDGE = isEdgeToEdge();

/**
 * @deprecated Please use the exported variable `reanimatedVersion` instead.
 * @returns `false` in Reanimated 4, `true` in Reanimated 3, doesn't exist in
 *   Reanimated 2 or 1
 */
export const isReanimated3 = () => {
  logger.warn('The `isReanimated3` function is deprecated. Please use the exported variable `reanimatedVersion` instead.');
  return false;
};

// Superseded by check in `/src/threads.ts`.
// Used by `react-navigation` to detect if using Reanimated 2 or 3.
/**
 * @deprecated Please use the exported variable `reanimatedVersion` instead.
 * @returns `false` in Reanimated 4, `true` in Reanimated 3, doesn't exist in
 *   Reanimated 2 or 1
 */
export const isConfigured = isReanimated3;
export function getViewProp(viewTag, propName, component) {
  if (!component) {
    throw new ReanimatedError('Function `getViewProp` requires a component to be passed as an argument on Fabric.');
  }

  // eslint-disable-next-line @typescript-eslint/no-misused-promises
  return new Promise((resolve, reject) => {
    return ReanimatedModule.getViewProp(viewTag, propName, component, result => {
      if (typeof result === 'string' && result.slice(0, 6) === 'error:') {
        // eslint-disable-next-line @typescript-eslint/prefer-promise-reject-errors
        reject(result);
      } else {
        resolve(result);
      }
    });
  });
}
function getSensorContainer() {
  if (!global.__sensorContainer) {
    global.__sensorContainer = new SensorContainer();
  }
  return global.__sensorContainer;
}
export function registerEventHandler(eventHandler, eventName, emitterReactTag = -1) {
  function handleAndFlushAnimationFrame(eventTimestamp, event) {
    'worklet';

    // TODO: Fix this and don't call `__flushAnimationFrame` here.
    global.__frameTimestamp = eventTimestamp;
    eventHandler(event);
    global.__flushAnimationFrame(eventTimestamp);
    global.__frameTimestamp = undefined;
  }
  return ReanimatedModule.registerEventHandler(createSerializable(handleAndFlushAnimationFrame), eventName, emitterReactTag);
}
export function unregisterEventHandler(id) {
  return ReanimatedModule.unregisterEventHandler(id);
}
export function subscribeForKeyboardEvents(eventHandler, options) {
  // TODO: this should really go with the same code path as other events, that is
  // via registerEventHandler. For now we are copying the code from there.
  function handleAndFlushAnimationFrame(state, height) {
    'worklet';

    // TODO: Fix this and don't call `__flushAnimationFrame` here.
    const now = global._getAnimationTimestamp();
    global.__frameTimestamp = now;
    eventHandler(state, height);
    global.__flushAnimationFrame(now);
    global.__frameTimestamp = undefined;
  }
  if (__DEV__) {
    controlEdgeToEdgeValues({
      isStatusBarTranslucentAndroid: options.isStatusBarTranslucentAndroid,
      isNavigationBarTranslucentAndroid: options.isNavigationBarTranslucentAndroid
    });
  }
  return ReanimatedModule.subscribeForKeyboardEvents(createSerializable(handleAndFlushAnimationFrame), EDGE_TO_EDGE || (options.isStatusBarTranslucentAndroid ?? false), EDGE_TO_EDGE || (options.isNavigationBarTranslucentAndroid ?? false));
}
export function unsubscribeFromKeyboardEvents(listenerId) {
  return ReanimatedModule.unsubscribeFromKeyboardEvents(listenerId);
}
export function registerSensor(sensorType, config, eventHandler) {
  const sensorContainer = getSensorContainer();
  return sensorContainer.registerSensor(sensorType, config, createSerializable(eventHandler));
}
export function initializeSensor(sensorType, config) {
  const sensorContainer = getSensorContainer();
  return sensorContainer.initializeSensor(sensorType, config);
}
export function unregisterSensor(sensorId) {
  const sensorContainer = getSensorContainer();
  return sensorContainer.unregisterSensor(sensorId);
}

/**
 * @deprecated This function no longer has any effect in Reanimated and will be
 *   removed in the future.
 */
export function enableLayoutAnimations(_flag, _isCallByUser = true) {
  logger.warn('`enableLayoutAnimations` is deprecated and will be removed in the future.');
}
export function configureLayoutAnimationBatch(layoutAnimationsBatch) {
  ReanimatedModule.configureLayoutAnimationBatch(layoutAnimationsBatch);
}
export function setShouldAnimateExitingForTag(viewTag, shouldAnimate) {
  ReanimatedModule.setShouldAnimateExitingForTag(viewTag, shouldAnimate);
}
//# sourceMappingURL=core.js.map