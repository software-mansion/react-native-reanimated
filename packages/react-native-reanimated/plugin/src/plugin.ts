import type { NodePath, PluginItem } from '@babel/core';
import type {
  CallExpression,
  ClassDeclaration,
  JSXAttribute,
  ObjectExpression,
  Program,
} from '@babel/types';
import {
  processCalleesAutoworkletizableCallbacks,
  processIfAutoworkletizableCallback,
} from './autoworkletization';
import { processIfWorkletClass } from './class';
import { processIfWorkletContextObject } from './contextObject';
import { processIfWorkletFile } from './file';
import { initializeState } from './globals';
import { processInlineStylesWarning } from './inlineStylesWarning';
import type { ReanimatedPluginPass } from './types';
import { WorkletizableFunction } from './types';
import { substituteWebCallExpression } from './webOptimization';
import { processIfWithWorkletDirective } from './workletSubstitution';

module.exports = function (): PluginItem {
  function runWithTaggedExceptions(fun: () => void) {
    try {
      fun();
    } catch (e) {
      throw new Error(`[Reanimated] Babel plugin exception: ${e as string}`);
    }
  }

  return {
    name: 'reanimated',

    pre(this: ReanimatedPluginPass) {
      runWithTaggedExceptions(() => {
        initializeState(this);
      });
    },
    visitor: {
      CallExpression: {
        enter(path: NodePath<CallExpression>, state: ReanimatedPluginPass) {
          runWithTaggedExceptions(() => {
            processCalleesAutoworkletizableCallbacks(path, state);
            if (state.opts.substituteWebPlatformChecks) {
              substituteWebCallExpression(path);
            }
          });
        },
      },
      [WorkletizableFunction]: {
        enter(
          path: NodePath<WorkletizableFunction>,
          state: ReanimatedPluginPass
        ) {
          runWithTaggedExceptions(() => {
            processIfWithWorkletDirective(path, state) ||
              processIfAutoworkletizableCallback(path, state);
          });
        },
      },
      ObjectExpression: {
        enter(path: NodePath<ObjectExpression>, state: ReanimatedPluginPass) {
          runWithTaggedExceptions(() => {
            processIfWorkletContextObject(path, state);
          });
        },
      },
      ClassDeclaration: {
        enter(path: NodePath<ClassDeclaration>, state: ReanimatedPluginPass) {
          runWithTaggedExceptions(() => {
            processIfWorkletClass(path, state);
          });
        },
      },
      Program: {
        enter(path: NodePath<Program>, state: ReanimatedPluginPass) {
          runWithTaggedExceptions(() => {
            processIfWorkletFile(path, state);
          });
        },
      },
      JSXAttribute: {
        enter(path: NodePath<JSXAttribute>, state: ReanimatedPluginPass) {
          runWithTaggedExceptions(() =>
            processInlineStylesWarning(path, state)
          );
        },
      },
    },
  };
};
