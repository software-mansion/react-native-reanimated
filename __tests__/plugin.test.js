import plugin from '../plugin';
import { transform } from '@babel/core';
import traverse from '@babel/traverse';

function runPlugin(input, opts = {}) {
  return transform(input, {
    filename: 'jest tests fixture',
    compact: false,
    plugins: [plugin],
    ...opts,
  });
}

describe('babel plugin', () => {
  it('transforms', () => {
    const input = `
      import Animated, {
        useAnimatedStyle,
        useSharedValue,
      } from 'react-native-reanimated';
      
      function Box() {
        const offset = useSharedValue(0);
      
        const animatedStyles = useAnimatedStyle(() => {
          return {
            transform: [{ translateX: offset.value * 255 }],
          };
        });
      
        return (
          <>
            <Animated.View style={[styles.box, animatedStyles]} />
            <Button onPress={() => (offset.value = Math.random())} title="Move" />
          </>
        );
      }
    `;

    const { code } = runPlugin(input);
    expect(code).toMatchSnapshot();
  });

  it('supports default ES6 style imports', () => {
    const input = `
      import * as Reanimated from 'react-native-reanimated';

      function Box() {
        const offset = Reanimated.useSharedValue(0);
      
        const animatedStyles = Reanimated.useAnimatedStyle(() => {
          return {
            transform: [{ translateX: offset.value * 255 }],
          };
        });
      
        return (
          <>
            <Animated.View style={[styles.box, animatedStyles]} />
            <Button onPress={() => (offset.value = Math.random())} title="Move" />
          </>
        );
      }
    `;

    const { code } = runPlugin(input);
    expect(code).toContain('_closure');
  });

  it("doesn't transform functions without 'worklet' directive", () => {
    const input = `
      function f(x) {
        return x + 2;
      }
    `;

    const { code } = runPlugin(input);
    expect(code).not.toContain('_f.__workletHash');
  });

  it('removes comments from worklets', () => {
    const input = `
      const f = () => {
        'worklet';
        // some comment
        /*
        * other comment
        */
        return true;
      };
    `;

    const { code } = runPlugin(input);
    expect(code).not.toContain('some comment');
    expect(code).not.toContain('other comment');
  });

  it('removes "worklet"; directive from worklets', () => {
    const input = `
      function foo(x) {
        "worklet"; // prettier-ignore
        return x + 2;
      }
    `;

    const { code } = runPlugin(input);
    expect(code).not.toContain('\\"worklet\\";');
  });

  it("removes 'worklet'; directive from worklets", () => {
    const input = `
      function foo(x) {
        'worklet'; // prettier-ignore
        return x + 2;
      }
    `;

    const { code } = runPlugin(input);
    expect(code).not.toContain("'worklet';");
  });

  it("doesn't transform string literals", () => {
    const input = `
      function foo(x) {
        'worklet';
        const bar = 'worklet'; // prettier-ignore
        const baz = "worklet"; // prettier-ignore
      }
    `;

    const { code } = runPlugin(input);
    expect(code).toMatchSnapshot();
  });

  it('captures worklets environment', () => {
    const input = `
      const x = 5;

      const objX = { x };
      
      function f() {
        'worklet';
        return { res: x + objX.x };
      }
    `;

    const { code } = runPlugin(input);
    expect(code).toMatchSnapshot();
  });

  it("doesn't capture globals", () => {
    const input = `
      function f() {
        'worklet';
        console.log('test');
      }
    `;

    const { code, ast } = runPlugin(input, { ast: true });
    let closureBindings;
    traverse(ast, {
      enter(path) {
        if (
          path.isAssignmentExpression() &&
          path.node.left.property.name === '_closure'
        ) {
          closureBindings = path.node.right.properties;
        }
      },
    });
    expect(closureBindings).toEqual([]);
    expect(code).toMatchSnapshot();
  });

  // functions

  it('workletizes FunctionDeclaration', () => {
    const input = `
      function foo(x) {
        'worklet';
        return x + 2;
      }
    `;

    const { code } = runPlugin(input);
    expect(code).toContain('_f.__workletHash');
    expect(code).not.toContain('\\"worklet\\";');
    expect(code).toMatchSnapshot();
  });

  it('workletizes ArrowFunctionExpression', () => {
    const input = `
      const foo = (x) => {
        'worklet';
        return x + 2;
      };
    `;

    const { code } = runPlugin(input);
    expect(code).toContain('_f.__workletHash');
    expect(code).not.toContain('\\"worklet\\";');
    expect(code).toMatchSnapshot();
  });

  it('workletizes unnamed FunctionExpression', () => {
    const input = `
      const foo = function (x) {
        'worklet';
        return x + 2;
      };
    `;

    const { code } = runPlugin(input);
    expect(code).toContain('_f.__workletHash');
    expect(code).not.toContain('\\"worklet\\";');
    expect(code).toMatchSnapshot();
  });

  it('workletizes named FunctionExpression', () => {
    const input = `
      const foo = function foo(x) {
        'worklet';
        return x + 2;
      };
    `;

    const { code } = runPlugin(input);
    expect(code).toContain('_f.__workletHash');
    expect(code).not.toContain('\\"worklet\\";');
    expect(code).toMatchSnapshot();
  });

  // class methods

  it('workletizes instance method', () => {
    const input = `
      class Foo {
        bar(x) {
          'worklet';
          return x + 2;
        }
      }
    `;

    const { code } = runPlugin(input);
    expect(code).toContain('_f.__workletHash');
    expect(code).not.toContain('\\"worklet\\";');
    expect(code).toMatchSnapshot();
  });

  it('workletizes static method', () => {
    const input = `
      class Foo {
        static bar(x) {
          'worklet';
          return x + 2;
        }
      }
    `;

    const { code } = runPlugin(input);
    expect(code).toContain('_f.__workletHash');
    expect(code).not.toContain('\\"worklet\\";');
    expect(code).toMatchSnapshot();
  });

  it('workletizes getter', () => {
    const input = `
      class Foo {
        get bar() {
          'worklet';
          return x + 2;
        }
      }
    `;

    const { code } = runPlugin(input);
    expect(code).toContain('_f.__workletHash');
    expect(code).not.toContain('\\"worklet\\";');
    expect(code).toMatchSnapshot();
  });

  // function hooks

  it('workletizes hook wrapped ArrowFunctionExpression automatically', () => {
    const input = `
      const animatedStyle = useAnimatedStyle(() => ({
        width: 50,
      }));
    `;

    const { code } = runPlugin(input);
    expect(code).toContain('_f.__workletHash');
    expect(code).toMatchSnapshot();
  });

  it('workletizes hook wrapped unnamed FunctionExpression automatically', () => {
    const input = `
      const animatedStyle = useAnimatedStyle(function () {
        return {
          width: 50,
        };
      });
    `;

    const { code } = runPlugin(input);
    expect(code).toContain('_f.__workletHash');
    expect(code).toMatchSnapshot();
  });

  it('workletizes hook wrapped named FunctionExpression automatically', () => {
    const input = `
      const animatedStyle = useAnimatedStyle(function foo() {
        return {
          width: 50,
        };
      });
    `;

    const { code } = runPlugin(input);
    expect(code).toContain('_f.__workletHash');
    expect(code).toMatchSnapshot();
  });

  // object hooks

  it('workletizes object hook wrapped ArrowFunctionExpression automatically', () => {
    const input = `
      useAnimatedGestureHandler({
        onStart: (event) => {
          console.log(event);
        },
      });
    `;

    const { code } = runPlugin(input);
    expect(code).toContain('_f.__workletHash');
    expect(code).toMatchSnapshot();
  });

  it('workletizes object hook wrapped unnamed FunctionExpression automatically', () => {
    const input = `
      useAnimatedGestureHandler({
        onStart: function (event) {
          console.log(event);
        },
      });
    `;

    const { code } = runPlugin(input);
    expect(code).toContain('_f.__workletHash');
    expect(code).toMatchSnapshot();
  });

  it('workletizes object hook wrapped named FunctionExpression automatically', () => {
    const input = `
      useAnimatedGestureHandler({
        onStart: function onStart(event) {
          console.log(event);
        },
      });
    `;

    const { code } = runPlugin(input);
    expect(code).toContain('_f.__workletHash');
    expect(code).toMatchSnapshot();
  });

  it('workletizes object hook wrapped ObjectMethod automatically', () => {
    const input = `
      useAnimatedGestureHandler({
        onStart(event) {
          console.log(event);
        },
      });
    `;

    const { code } = runPlugin(input);
    expect(code).toContain('_f.__workletHash');
    expect(code).toMatchSnapshot();
  });

  it('supports empty object in hooks', () => {
    const input = `
      useAnimatedGestureHandler({});
    `;

    runPlugin(input);
  });

  it('transforms each object property in hooks', () => {
    const input = `
      useAnimatedGestureHandler({
        onStart: () => {},
        onUpdate: () => {},
        onEnd: () => {},
      });
    `;

    const { code } = runPlugin(input);
    expect(code).toMatch(/^(.*)(_f\.__workletHash(.*)){3}$/s);
  });

  // React Native Gesture Handler

  it('workletizes possibly chained gesture object callback functions automatically', () => {
    const input = `
      import { Gesture } from 'react-native-gesture-handler';

      const foo = Gesture.Tap()
        .numberOfTaps(2)
        .onBegin(() => {
          console.log('onBegin');
        })
        .onStart((_event) => {
          console.log('onStart');
        })
        .onEnd((_event, _success) => {
          console.log('onEnd');
        });
    `;

    const { code } = runPlugin(input);
    expect(code).toMatchSnapshot();
  });

  it("doesn't transform standard callback functions", () => {
    const input = `
      const foo = Something.Tap().onEnd((_event, _success) => {
        console.log('onEnd');
      });
    `;

    const { code } = runPlugin(input);
    expect(code).toMatchSnapshot();
  });

  it('transforms spread operator in worklets for arrays', () => {
    const input = `
      function foo() {
        'worklet';
        const bar = [4, 5];
        const baz = [1, ...[2, 3], ...bar];
      }
    `;

    const { code } = runPlugin(input);
    expect(code).toMatchSnapshot();
  });

  it('transforms spread operator in worklets for objects', () => {
    const input = `
      function foo() {
        'worklet';
        const bar = {d: 4, e: 5};
        const baz = { a: 1, ...{ b: 2, c: 3 }, ...bar };
      }
    `;

    const { code } = runPlugin(input);
    expect(code).toMatchSnapshot();
  });

  it('transforms spread operator in worklets for function arguments', () => {
    const input = `
      function foo(...args) {
        'worklet';
        console.log(args);
      }
    `;

    const { code } = runPlugin(input);
    expect(code).toMatchSnapshot();
  });

  it('transforms spread operator in worklets for function calls', () => {
    const input = `
      function foo(arg) {
        'worklet';
        console.log(...arg);
      }
    `;

    const { code } = runPlugin(input);
    expect(code).toMatchSnapshot();
  });
});
