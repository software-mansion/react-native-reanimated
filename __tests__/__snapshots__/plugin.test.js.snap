// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel plugin captures worklets environment 1`] = `
"var x = 5;
var objX = {
  x: x
};

var f = function () {
  var _f = function _f() {
    return {
      res: x + objX.x
    };
  };

  _f._closure = {
    x: x,
    objX: {
      x: objX.x
    }
  };
  _f.asString = \\"function f(){const{x,objX}=jsThis._closure;{return{res:x+objX.x};}}\\";
  _f.__workletHash = 10184269015616;
  _f.__location = \\"${ process.cwd() }/jest tests fixture (6:6)\\";

  global.__reanimatedWorkletInit(_f);

  return _f;
}();"
`;

exports[`babel plugin doesn't capture globals 1`] = `
"var f = function () {
  var _f = function _f() {
    console.log('test');
  };

  _f._closure = {};
  _f.asString = \\"function f(){console.log('test');}\\";
  _f.__workletHash = 13298016111221;
  _f.__location = \\"${ process.cwd() }/jest tests fixture (2:6)\\";

  global.__reanimatedWorkletInit(_f);

  return _f;
}();"
`;

exports[`babel plugin doesn't transform standard callback functions 1`] = `
"var foo = Something.Tap().onEnd(function (_event, _success) {
  console.log('onEnd');
});"
`;

exports[`babel plugin doesn't transform string literals 1`] = `
"var foo = function () {
  var _f = function _f(x) {
    var bar = 'worklet';
    var baz = \\"worklet\\";
  };

  _f._closure = {};
  _f.asString = \\"function foo(x){const bar='worklet';const baz=\\\\\\"worklet\\\\\\";}\\";
  _f.__workletHash = 9810417751380;
  _f.__location = \\"${ process.cwd() }/jest tests fixture (2:6)\\";

  global.__reanimatedWorkletInit(_f);

  return _f;
}();"
`;

exports[`babel plugin transforms 1`] = `
"var _interopRequireWildcard = require(\\"@babel/runtime/helpers/interopRequireWildcard\\");

var _reactNativeReanimated = _interopRequireWildcard(require(\\"react-native-reanimated\\"));

function Box() {
  var offset = (0, _reactNativeReanimated.useSharedValue)(0);
  var animatedStyles = (0, _reactNativeReanimated.useAnimatedStyle)(function () {
    var _f = function _f() {
      return {
        transform: [{
          translateX: offset.value * 255
        }]
      };
    };

    _f._closure = {
      offset: offset
    };
    _f.asString = \\"function _f(){const{offset}=jsThis._closure;{return{transform:[{translateX:offset.value*255}]};}}\\";
    _f.__workletHash = 7114514849439;
    _f.__location = \\"${ process.cwd() }/jest tests fixture (10:48)\\";
    _f.__optimalization = 3;

    global.__reanimatedWorkletInit(_f);

    return _f;
  }());
  return React.createElement(React.Fragment, null, React.createElement(_reactNativeReanimated.default.View, {
    style: [styles.box, animatedStyles]
  }), React.createElement(Button, {
    onPress: function onPress() {
      return offset.value = Math.random();
    },
    title: \\"Move\\"
  }));
}"
`;

exports[`babel plugin workletizes ArrowFunctionExpression 1`] = `
"var foo = function () {
  var _f = function _f(x) {
    return x + 2;
  };

  _f._closure = {};
  _f.asString = \\"function _f(x){return x+2;}\\";
  _f.__workletHash = 11411090164019;
  _f.__location = \\"${ process.cwd() }/jest tests fixture (2:18)\\";

  global.__reanimatedWorkletInit(_f);

  return _f;
}();"
`;

exports[`babel plugin workletizes FunctionDeclaration 1`] = `
"var foo = function () {
  var _f = function _f(x) {
    return x + 2;
  };

  _f._closure = {};
  _f.asString = \\"function foo(x){return x+2;}\\";
  _f.__workletHash = 4679479961836;
  _f.__location = \\"${ process.cwd() }/jest tests fixture (2:6)\\";

  global.__reanimatedWorkletInit(_f);

  return _f;
}();"
`;

exports[`babel plugin workletizes getter 1`] = `
"var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\");

var _classCallCheck2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/classCallCheck\\"));

var _createClass2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/createClass\\"));

var Foo = function () {
  function Foo() {
    (0, _classCallCheck2.default)(this, Foo);
  }

  (0, _createClass2.default)(Foo, [{
    key: \\"bar\\",
    get: function () {
      var _f = function _f() {
        return x + 2;
      };

      _f._closure = {
        x: x
      };
      _f.asString = \\"function get(){const{x}=jsThis._closure;{return x+2;}}\\";
      _f.__workletHash = 2468276954688;
      _f.__location = \\"${ process.cwd() }/jest tests fixture\\";

      global.__reanimatedWorkletInit(_f);

      return _f;
    }()
  }]);
  return Foo;
}();"
`;

exports[`babel plugin workletizes hook wrapped ArrowFunctionExpression automatically 1`] = `
"var animatedStyle = useAnimatedStyle(function () {
  var _f = function _f() {
    return {
      width: 50
    };
  };

  _f._closure = {};
  _f.asString = \\"function _f(){width:50;}\\";
  _f.__workletHash = 5067980299705;
  _f.__location = \\"${ process.cwd() }/jest tests fixture (2:45)\\";
  _f.__optimalization = 3;

  global.__reanimatedWorkletInit(_f);

  return _f;
}());"
`;

exports[`babel plugin workletizes hook wrapped named FunctionExpression automatically 1`] = `
"var animatedStyle = useAnimatedStyle(function () {
  var _f = function _f() {
    return {
      width: 50
    };
  };

  _f._closure = {};
  _f.asString = \\"function foo(){return{width:50};}\\";
  _f.__workletHash = 6275510763626;
  _f.__location = \\"${ process.cwd() }/jest tests fixture (2:45)\\";
  _f.__optimalization = 3;

  global.__reanimatedWorkletInit(_f);

  return _f;
}());"
`;

exports[`babel plugin workletizes hook wrapped unnamed FunctionExpression automatically 1`] = `
"var animatedStyle = useAnimatedStyle(function () {
  var _f = function _f() {
    return {
      width: 50
    };
  };

  _f._closure = {};
  _f.asString = \\"function _f(){return{width:50};}\\";
  _f.__workletHash = 9756190407413;
  _f.__location = \\"${ process.cwd() }/jest tests fixture (2:45)\\";
  _f.__optimalization = 3;

  global.__reanimatedWorkletInit(_f);

  return _f;
}());"
`;

exports[`babel plugin workletizes instance method 1`] = `
"var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\");

var _classCallCheck2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/classCallCheck\\"));

var _createClass2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/createClass\\"));

var Foo = function () {
  function Foo() {
    (0, _classCallCheck2.default)(this, Foo);
  }

  (0, _createClass2.default)(Foo, [{
    key: \\"bar\\",
    value: function () {
      var _f = function _f(x) {
        return x + 2;
      };

      _f._closure = {};
      _f.asString = \\"function bar(x){return x+2;}\\";
      _f.__workletHash = 16974800582491;
      _f.__location = \\"${ process.cwd() }/jest tests fixture\\";

      global.__reanimatedWorkletInit(_f);

      return _f;
    }()
  }]);
  return Foo;
}();"
`;

exports[`babel plugin workletizes named FunctionExpression 1`] = `
"var foo = function () {
  var _f = function _f(x) {
    return x + 2;
  };

  _f._closure = {};
  _f.asString = \\"function foo(x){return x+2;}\\";
  _f.__workletHash = 4679479961836;
  _f.__location = \\"${ process.cwd() }/jest tests fixture (2:18)\\";

  global.__reanimatedWorkletInit(_f);

  return _f;
}();"
`;

exports[`babel plugin workletizes object hook wrapped ArrowFunctionExpression automatically 1`] = `
"useAnimatedGestureHandler({
  onStart: function () {
    var _f = function _f(event) {
      console.log(event);
    };

    _f._closure = {};
    _f.asString = \\"function _f(event){console.log(event);}\\";
    _f.__workletHash = 2164830539996;
    _f.__location = \\"${ process.cwd() }/jest tests fixture (3:17)\\";

    global.__reanimatedWorkletInit(_f);

    return _f;
  }()
});"
`;

exports[`babel plugin workletizes object hook wrapped ObjectMethod automatically 1`] = `
"useAnimatedGestureHandler({
  onStart: function () {
    var _f = function _f(event) {
      console.log(event);
    };

    _f._closure = {};
    _f.asString = \\"function onStart(event){console.log(event);}\\";
    _f.__workletHash = 338158776260;
    _f.__location = \\"${ process.cwd() }/jest tests fixture (3:8)\\";

    global.__reanimatedWorkletInit(_f);

    return _f;
  }()
});"
`;

exports[`babel plugin workletizes object hook wrapped named FunctionExpression automatically 1`] = `
"useAnimatedGestureHandler({
  onStart: function () {
    var _f = function _f(event) {
      console.log(event);
    };

    _f._closure = {};
    _f.asString = \\"function onStart(event){console.log(event);}\\";
    _f.__workletHash = 338158776260;
    _f.__location = \\"${ process.cwd() }/jest tests fixture (3:17)\\";

    global.__reanimatedWorkletInit(_f);

    return _f;
  }()
});"
`;

exports[`babel plugin workletizes object hook wrapped unnamed FunctionExpression automatically 1`] = `
"useAnimatedGestureHandler({
  onStart: function () {
    var _f = function _f(event) {
      console.log(event);
    };

    _f._closure = {};
    _f.asString = \\"function _f(event){console.log(event);}\\";
    _f.__workletHash = 2164830539996;
    _f.__location = \\"${ process.cwd() }/jest tests fixture (3:17)\\";

    global.__reanimatedWorkletInit(_f);

    return _f;
  }()
});"
`;

exports[`babel plugin workletizes possibly chained gesture object callback functions automatically 1`] = `
"var _reactNativeGestureHandler = require(\\"react-native-gesture-handler\\");

var foo = _reactNativeGestureHandler.Gesture.Tap().numberOfTaps(2).onBegin(function () {
  var _f = function _f() {
    console.log('onBegin');
  };

  _f._closure = {};
  _f.asString = \\"function _f(){console.log('onBegin');}\\";
  _f.__workletHash = 13662490049850;
  _f.__location = \\"${ process.cwd() }/jest tests fixture (6:17)\\";

  global.__reanimatedWorkletInit(_f);

  return _f;
}()).onStart(function () {
  var _f = function _f(_event) {
    console.log('onStart');
  };

  _f._closure = {};
  _f.asString = \\"function _f(_event){console.log('onStart');}\\";
  _f.__workletHash = 16334902412526;
  _f.__location = \\"${ process.cwd() }/jest tests fixture (9:17)\\";

  global.__reanimatedWorkletInit(_f);

  return _f;
}()).onEnd(function () {
  var _f = function _f(_event, _success) {
    console.log('onEnd');
  };

  _f._closure = {};
  _f.asString = \\"function _f(_event,_success){console.log('onEnd');}\\";
  _f.__workletHash = 4053780716017;
  _f.__location = \\"${ process.cwd() }/jest tests fixture (12:15)\\";

  global.__reanimatedWorkletInit(_f);

  return _f;
}());"
`;

exports[`babel plugin workletizes static method 1`] = `
"var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\");

var _classCallCheck2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/classCallCheck\\"));

var _createClass2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/createClass\\"));

var Foo = function () {
  function Foo() {
    (0, _classCallCheck2.default)(this, Foo);
  }

  (0, _createClass2.default)(Foo, null, [{
    key: \\"bar\\",
    value: function () {
      var _f = function _f(x) {
        return x + 2;
      };

      _f._closure = {};
      _f.asString = \\"function bar(x){return x+2;}\\";
      _f.__workletHash = 16974800582491;
      _f.__location = \\"${ process.cwd() }/jest tests fixture\\";

      global.__reanimatedWorkletInit(_f);

      return _f;
    }()
  }]);
  return Foo;
}();"
`;

exports[`babel plugin workletizes unnamed FunctionExpression 1`] = `
"var foo = function () {
  var _f = function _f(x) {
    return x + 2;
  };

  _f._closure = {};
  _f.asString = \\"function _f(x){return x+2;}\\";
  _f.__workletHash = 11411090164019;
  _f.__location = \\"${ process.cwd() }/jest tests fixture (2:18)\\";

  global.__reanimatedWorkletInit(_f);

  return _f;
}();"
`;
